frontend/src/metabase/visualizations/components/LineAreaBarChart.jsx

```diff
@@ -35,7 +35,7 @@ const isDimensionDimensionMetric = (cols, strict = true) =>
     isMetric(cols[2])
 
 const isDimensionMetricMetric = (cols, strict = true) =>
```

Comment:
We use the same functions for determining "sensible" chart types and determining which type of multiseries chart (if any) to transform it to. `strict = true` causes each type to check that it has exactly the expected number and type of columns, and no more. The problem was D-M was being displayed as D-M-M, causing the multiseries header to be shown.